gpustat/__main__.py

```diff
@@ -11,12 +11,16 @@
 from .core import GPUStatCollection
 
 
-def print_gpustat(json=False, debug=False, **kwargs):
+def print_gpustat(json=False, debug=False, gpu_stat=None, **kwargs):
     '''
     Display the GPU query results into standard output.
     '''
+
     try:
-        gpu_stats = GPUStatCollection.new_query()
+        if gpu_stat is None:
+            gpu_stat = GPUStatCollection()
+        else:
+            gpu_stat.update()
     except Exception as e:
         sys.stderr.write('Error on querying NVIDIA devices.'
```

Comment:
The error catching workflow is complicated for me, can we just throw the raw error messages instead of the current one.

Reply:
For example, if an error happens during the loop, we can simply throw it out.

Reply:
The error catching requires all the **NVML** API calls to be wrapped with a long `try-catch` block.